src/item.cpp

```diff
@@ -4720,7 +4721,8 @@ int item::reload_option::moves() const
 
 void item::reload_option::qty( long val )
 {
-    if( ammo->is_magazine() || target->has_flag( "RELOAD_ONE" )) {
+    if( !ammo->has_flag( "SPEEDLOADER" ) &&
+        ( ammo->is_magazine() || target->has_flag( "RELOAD_ONE" ) ) ) {
```

Comment:
This looks wrong, though I didn't check how it works.
You're reloading a gun with one magazine, a stack of ammo, or a "stack of speedloaders". For speedloaders, the qty should probably always be 1, actual ammo count should only really matter for display (already handled for magazines) and after reload, when the gun gets the ammo.

Reply:
There's a bunch of code that cares whether the gun in question has an internal or detachable magazine.  The internal_magazine code tracks the number of rounds available to reload, and it needs _qty to have the right value.

Reply:
But didn't you special case reloading with speedloaders to require 0 ammo?
To me it sounds like it would be much simpler to always treat speedloaders as qty 1 than to try to combine treatment of ammo (charges) with that of magazines (always exactly one) in different cases.

Reply:
> But didn't you special case reloading with speedloaders to require 0 ammo?

I did not, the requirement you are probably referencing is that the target gun (or magazine, I should add a detachable magazine speedloader as an example...) must have no ammunition, I didn't want to mess around with forcing an unload, when the player can just do it themselves, that reminds me I need to prompt about that.